src/coreclr/jit/gentree.cpp

```diff
@@ -32529,45 +32529,6 @@ GenTree* Compiler::gtFoldExprHWIntrinsic(GenTreeHWIntrinsic* tree)
     }
 #endif // FEATURE_MASKED_HW_INTRINSICS && TARGET_XARCH
 
-    switch (ni)
```

Comment:
The whole `gtFoldExprHWIntrinsic` method was refactored a bit to ensure we could keep the new checks together with the existing logic.

The main change was that this was moved down to live with the rest of the `opCount == 3` handling and to reuse the existing "is one of the operands constant" checks.